File: mod/default/goto.ts
//:goto
const goto = ((segment = args.split(' ')[0]): returnAction => {
    if (!registry[segment] && registry[segment] !== 0) {
        console.log('CRITICAL FAILURE: Couldn\'t find segment \'' + segment + '\' to run goto at line ' + (i + 1) + '!')
        return returnAction.CRITICAL
    }
    stack[stack.length - 1] = i + 1
    stack.push(registry[segment])

    i = registry[segment]
    return returnAction.PEACEFUL
})

File: nim/mod/default/goto.nim
template mod_default_goto() =
    proc goto(): returnAction =
        expect("1")

        var segment {.inject.} = args.split(' ')[0]
        if(not registry.hasKey(segment) and registry[segment].n != 0):
            SetFailure(fmt"Couldn't find segment '{segment}' to run goto at line {i.n + 1}")
            return returnAction.CRITICAL

        stack[stack.len - 1].n = i.n + 1
        stack.add(registry[segment])

        i = registry[segment]
        return returnAction.CONTINUE

    internals["goto"] = goto

export mod_default_goto